apps/api/src/http/validators.rs

```diff
@@ -21,6 +21,27 @@ pub fn check_emote_name(value: impl AsRef<str>) -> bool {
 		.is_match(value.as_ref())
 }
 
+#[derive(Debug, Copy, Clone)]
+pub struct EmoteAliasValidator;
+
+impl CustomValidator<String> for EmoteAliasValidator {
+	fn check(&self, value: &String) -> Result<(), async_graphql::InputValueError<String>> {
+		if check_emote_alias(value) {
+			Ok(())
+		} else {
+			Err(async_graphql::InputValueError::custom("invalid emote alias"))
+		}
+	}
+}
+
+pub fn check_emote_alias(value: impl AsRef<str>) -> bool {
+	static REGEX: std::sync::OnceLock<regex::Regex> = std::sync::OnceLock::new();
+
+	REGEX
+		.get_or_init(|| regex::Regex::new(r"^[^\x00-\x20\u{00A0}\u{180E}\u{2000}-\u{200A}\u{202F}\u{205F}\u{3000}\u{E0000}]{1,100}$").unwrap())
+		.is_match(value.as_ref())
+}
```

Comment:
Imo a different regex shouldn't be used here, and instead the `check_emote_name` function should be used. There's no reason for inconsistency here.
https://github.com/SevenTV/SevenTV/blob/0d1e6edf8bc7b37d7731780e59fbd1ebd81b4383/apps/api/src/http/validators.rs#L16-L22

Reply:
As I said in my original comment, this allows channels to use Cyrillic and other non-Latin characters for their emote names, while keeping global moderated emote names Latin only.

Reply:
Why not alter the `check_emote_name` with the Cyrillic unicode blocks and any other you want added in instead? It makes no sense to have these be inconsistent.

Reply:
Because those names are moderated and 7TV doesn't have mods for every language. Also, I'd claim that leaving the actual names as Latin only makes all emotes easier to find.

Reply:
> Because those names are moderated and 7TV doesn't have mods for every language.

This has nothing to do with character validation.

> I'd claim that leaving the actual names as Latin only makes all emotes easier to find.

Agreed, but that's already not the case, so it's pointless to even consider this.

Reply:
I'll leave the decision to the 7tv team. But if I understood correctly, they previously disallowed other characters in emote names partly due to those reasons.

Reply:
Can you set this as the alias regex? `"^[\w\-():!+|.'?><\p{Emoji_Presentation}*$#]{1,100}$")`
Allowing anything except whitespace is a bit too much

Reply:
Seems to cover everything except localized punctuation, like the spanish `¿`. But I think those would rarely be used in aliases anyway. I'll change the regex to yours.